src/main/java/org/intellij/sdk/codesync/alerts/PricingAlerts.java

```diff
@@ -34,6 +34,20 @@ public static void setPlanLimitReached(Project project) {
         );
     }
 
+    public static void setPlanLimitReached(Project project, boolean privateRepoCountLimitReached) {
+        // We only want to show notification once every 5 minutes, I have implemented that using locks with an expiry of
+        // 5 minutes. So, skip the notification if lock is not acquired.
+        acquirePricingLock();
+        CommonUtils.invokeAndWait(
+                () -> {
+                    PricingAlertDialog pricingAlertDialog = new PricingAlertDialog(false, false, CODESYNC_PRICING_URL, project, privateRepoCountLimitReached);
```

Comment:
Why are we passing hardcoded false for `canAvailTrial`? Shouldn't this come from the API? `/v1/users/pricing/subscription` (I think)